orchagent/routeorch.cpp

```diff
@@ -564,6 +563,11 @@ bool RouteOrch::addNextHopGroup(IpAddresses ipAddresses)
 
     for (auto nhid: next_hop_ids)
     {
+        // skip next hop group member create for neighbor from down port
+        if (m_neighOrch->isNextHopFlagSet(nhopgroup_members_set[nhid], NHFLAGS_IFDOWN)) {
```

Comment:
The code was put intentionally at the end and not here. I'd like to understand the issue you were seeing more and I will also explain the logic why the code is at the end so lets discuss offline.

Reply:
Right,  I went through the current implementation.  The next_hop_id was created then removed, but the value is saved which is only used again when the removeNextHopGroup().  orch agent needs to have a copy of nexthop which it already has, but probably not the obsoleted next_hop_id.

It is possible I missed some of the key point. We may have offline discussion, then provide more details here.
btw, I'm glad you are back :)

Reply:
Unless I missed some other change in your diffs, with this one here I believe we will be leaking nexthop groups.